lib/index.js

```diff
@@ -15,17 +15,22 @@ module.exports = function(playlistURL, options) {
   options = options || {};
   var chunkReadahead = options.chunkReadahead || 3;
   var refreshInterval = options.refreshInterval || 600000; // 10 minutes
+  var requestOptions = options.requestOptions;
 
   var latestSegment;
-  var streamQueue = new Queue(function(segment, callback) {
+  var streamQueue = new Queue(function(getSegment, callback) {
+    var segment = getSegment();
     latestSegment = segment;
     segment.pipe(stream, { end: false });
     segment.on('error', callback);
     segment.on('end', callback);
   }, { concurrency: 1 });
 
   var requestQueue = new Queue(function(segmentURL, callback) {
-    streamQueue.push(miniget(urlResolve(playlistURL, segmentURL)), callback);
+    streamQueue.push(
+      function() { return miniget(urlResolve(playlistURL, segmentURL), requestOptions); },
```

Comment:
If this is wrapped in a function, the request won't be read ahead of time, which is the purpose of the `chunkReadhead` option.

Another way to catch the error here would be to move the `error` listener above, on line 25, to here.